ably.d.ts

```diff
@@ -363,7 +363,27 @@ declare namespace Types {
 	}
 
 	// Classes
-	class RestPromise extends Rest {
+	class RestBase {
+		constructor(options: Types.ClientOptions | string);
+		static Crypto: Types.Crypto;
+		static Message: Types.MessageStatic;
+		static PresenceMessage: Types.PresenceMessageStatic;
+	}
+
+	class RestCallbacks extends RestBase {
+		static Promise: typeof Types.RestPromise;
```

Comment:
why is this declared here?

Reply:
Because if I declare it in `RestBase`, then when `RealtimeBase` extends `RestBase` and overrides it with `static Promise: typeof Types.RealtimePromise;` etc., tsc objects with `--strict`

The alternative would be something like 
```
	class RestBaseBase {
		constructor(options: Types.ClientOptions | string);
		static Crypto: Types.Crypto;
		static Message: Types.MessageStatic;
		static PresenceMessage: Types.PresenceMessageStatic;
	}

	class RestBase extends RestBaseBase {
		static Promise: typeof Types.RestPromise;
		static Callbacks: typeof Types.RestCallbacks;
	}

	class RestCallbacks extends RestBase {
		...
	}

	class RestPromise extends RestBase {
		...
	}

	class RealtimeBase extends RestBaseBase {
		...
	}
```
which seemed a bit messy. But shrug, can switch to that if you prefer it

Reply:
No, stick with what you have